test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/function/BiPredicateTest.scala

```diff
@@ -0,0 +1,127 @@
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util.function
+
+import java.util.function.BiPredicate
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.scalajs.testsuite.utils.AssertThrows._
+
+class BiPredicateTest {
+  import BiPredicateTest._
+
+  private val largerThan10LessThan30 = makeBiPredicate[Int, Int](_ > 10, _ < 30)
+  private val even = makeBiPredicate[Int, Int](_ % 2 == 0, _ % 2 == 0)
+
+  private val throwingPredicate =
+    makeBiPredicate[Any, Any](x => throw new ThrowingPredicateException(x), _ => true)
+
+  private val dontCallPredicate =
+    makeBiPredicate[Any, Any](x => throw new AssertionError(s"dontCallPredicate.test($x)"), _ => true)
+
+  @Test def and(): Unit = {
+    // Truth table
+    val evenAndInRange = largerThan10LessThan30.and(even)
+
+    // In Both Range and Even
+    assertTrue(evenAndInRange.test(22, 22))
+    assertTrue(evenAndInRange.test(40, 22))
+
+    // In Both Range and Not Even
+    assertFalse(evenAndInRange.test(41, 22)) // Left Odd
+    assertFalse(evenAndInRange.test(42, 21)) // Right Odd
+    assertFalse(evenAndInRange.test(41, 21)) // Both Odd
+
+    // In One Range and Even
+    assertFalse(evenAndInRange.test(6, 6))   // Left Range false
+    assertFalse(evenAndInRange.test(42, 42)) // Right Range false
+    assertFalse(evenAndInRange.test(6, 42))  // Both Range false
+
+    // In One Range and Not Even
+    assertFalse(evenAndInRange.test(5, 5))   // Left Range false
+    assertFalse(evenAndInRange.test(41, 41)) // Right Range false
+    assertFalse(evenAndInRange.test(5, 41))  // Both Range false
+
+    // In One Range, One Not Even
+    assertFalse(evenAndInRange.test(21, 42)) // Left Range true, Left Not Even
+    assertFalse(evenAndInRange.test(42, 41)) // Left Range true, Right Not Even
+    assertFalse(evenAndInRange.test(5, 22))  // Right Range true, Left Not Even
+    assertFalse(evenAndInRange.test(5, 21))  // Right Range true, Right Not Even
+
+    // Short-circuit
+    assertFalse(largerThan10LessThan30.and(dontCallPredicate).test(5, 5))   // Left false, Right true
+    assertFalse(largerThan10LessThan30.and(dontCallPredicate).test(20, 40)) // Left true, Right false
+    assertFalse(largerThan10LessThan30.and(dontCallPredicate).test(5, 40))  // Both false
+
+    // Left false, Right true
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.and(dontCallPredicate).test(5, 5))
+
+    // Left true, Right false
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.and(dontCallPredicate).test(20, 40))
+
+    // Both false
+    assertThrows(classOf[ThrowingPredicateException],
+      throwingPredicate.and(dontCallPredicate).test(5, 40))
+
+  }
+
+  @Test def negate(): Unit = {
+    val notLeftLargerThan10AndRightLessThan30 = largerThan10LessThan30.negate()
+    assertTrue("Right false", notLeftLargerThan10AndRightLessThan30.test(20, 40))
```

Comment:
Please be consistent with comment or message style (the previous test used comments for this).

Reply:
The previous test had multiple comments that were identical, but had a parent comment that showed the difference.  I guess I did it this way here partly because I had a failure and the explicit message in the asserts are easier to track down which assert actually failed. That said, I think this is still a good point and I should probably update all to pass in the error message.